### ISSUE TITLE
Can't run hygen generator with-prompt

### ISSUE THREAD

**Issue by roNn23** [jondot/hygen#321]
Checked with different versions and on 3 different computers, I always ran into this error

    SyntaxError: Invalid left-hand side expression in prefix operation

when calling this:

    hygen generator with-prompt

**Comment by roNn23** [comment]
I think I figured out why it doesn't work.
After initializing with `hygen init self` several files are created, among others `_templates/generator/with-prompt/prompt.js` with the following content:

``js
---
to: _templates/<%= name %>/<%= action || 'new' %>/prompt.js
---

// see types of prompts:
// https://github.com/enquirer/enquirer/tree/master/examples
//
module.exports = [
  {
    type: 'input',
    name: 'message',
    message: "What's your message?"
  }
]
```

The first part is YAML and should not be in this JS file.

Furthermore, the command from the git doc does not seem to work. When I execute `hygen generator with-prompt mygen`, no new generator is created, but tries to access the previous generator under `_templates/generator/with-prompt/prompt.js`. And exactly this file is faulty which results in the above mentioned error.

In my eyes the documentation seems to be wrong here. If I simply create a generator manually using the example of the initial generator under `_templates/generator/with-prompt/`, everything seems to work.

**Comment by tonglil** [comment]
This was very confusing.

Also, renaming to `prompt.ejs.t` does not fix this for me:
```
tree
.
└── _templates
    ├── awesome-generator
    │   └── new
    │       └── hello.ejs.t
    └── generator
        ├── help
        │   └── index.ejs.t
        ├── new
        │   └── hello.ejs.t
        └── with-prompt
            ├── hello.ejs.t
            └── prompt.ejs.t
```

Commenting out does make it run for me.

Hygen v6.1.0

**Comment by jondot** [comment]
Did this issue reproduce?

**Comment by stavroslee** [comment]
I've had the same issue the way @roNn23 described it.  @roNn23's local fix worked for me (commenting out the YAML so the prompt works).  I guess the prompt.js code runs before the YAML is removed?

**Comment by virtuoushub** [comment]
I also ran into this issue, using @mimamuh 's [comment](https://github.com/jondot/hygen/issues/321#issuecomment-938449496) I was able to run `hygen generator with-prompt myNewGenerator --action new` by renaming `prompt.js` to `prompt.ejs.t`

I am using Hygen `6.1.0`

**Comment by kira1928** [comment]
Is this [commit](https://github.com/jondot/hygen/commit/3092de4a78e73dc369becebdf4adb2a7a22a4ce6) related?

**Comment by jondot** [comment]
yes, looks related, thanks for spotting this -- will be fixed for 6.1.1
